assets/css/main.css

```diff
@@ -246,46 +246,71 @@ html, body {
     list-style: none;
 }
 
-/* index thumbnails */
-.course-thumbnails .advert-thumbnail,
-.course-thumbnails .course-thumbnail {
-    height: 330px;
+
+/* index frontpage */
+
+.frontpage-panel {
+    margin: 3%;
+    margin-top: 30px;
+}
+
+.frontpage-panel .panel-heading {
+    border-bottom: solid 1px #eee;
+    font-weight: 500;
+}
+
+.frontpage .card {
+    margin-bottom: 20px;
+    border: solid 1px #ccc;
     position: relative;
+    height: 355px;
```

Comment:
This probably should be rechecked, assuming the button is removed.

Reply:
I tried to implement a smaller height, but then it has the same problem as the current front page, thus if the instance or course name is very long it does not fit well the text flows over. Or when trying to level up the footers of cards, the grid gets a bit messy on some screen widths.  

I also tried substituting the end of a longer instance name with dots or just hiding it, but Finnish words are very long and cutting (all of) them in the correct spots for that to look good on mobile and desktop was not really possible. 

Therefore I ended up increasing the height to accommodate longer course- and instance names. Alternatively we could add a restraint for the length of the names when teachers are setting them up or risk some of them flowing a bit over.

Reply:
One possibility is to allow some space and possibly use scroll for extra info. I don't know. Pretty hard to find a good solution for this... Would it be ok if different boxes would have different heights (I guess not)? Nevertheless, good arguments for the sice :+1:

Reply:
Has anyone tried this with `flex`? We could have a minimum height and all boxes grow if one needs it to be taller. 

It might mean a little re-writing styles if we lose the ready-made Bootstrap component.

Reply:
With the footer added as we talked during the daily on Friday, these look good to me at least now. The footer balances it out. Most of the courses or course-instance names on A+ seem to be longish, so in any case there is not much extra room. And the grid works nicely now too. In any case, in my opinion it is better than how they currently run over such as:

![image](https://user-images.githubusercontent.com/43036333/84651893-57c52a00-af13-11ea-8111-4266d6217536.png)

Reply:
> Has anyone tried this with flex? We could have a minimum height and all boxes grow if one needs it to be taller.

I think it would be a good solution. No one has been experienced enough to do that update with limited amount of time allocation. In the end, static sizes work most of the time. Assuming, it would be relatively simple to implement, then please poke me or markku.

Reply:
Here's an example I wrote in ~5 minutes; https://codepen.io/mtthwhggns/pen/ExPgoxq

Reply:
Seems a really good solution. The flex idea _can_ be implemented within the PR or can be left to a second PR.

Reply:
A flex-box would probably be a good solution :) I prefer leaving it to another pull request, as:
- The code now works and looks good and is in my opinion an improvement to the earlier. 
- The appearance changes from the point of a user would be minimal, even though that would be implemented later on, the grid-changes happen in the background, so it wouldn't confuse the users.
- The frontpage inherits the content block from a base-template, so it might need a little rework there and might not be a five-minute-fix. Just a hunch based on working with the grid of the front page :D

Reply:
Can you create an issue about that too? You can assign it to you, link it here and you can then resolve this conversation, thanks.

Reply:
Sure!

Reply:
Here is the issue for this: https://github.com/apluslms/a-plus/issues/583